### PR TITLE
feat(perf): Add min max exclusive time filters to span stats endpoint

### PR DESCRIPTION
**Pull request by udameli**
This PR adds an ability to add extra filters, `min_exclusive_time` and `max_exclusive_time`, so that users are able to view span stats for the selected population. This supplements histogram zooming feature, so in the span details page users are able to view the stats specific to the selected population.

Fixes PERF-1520

### CODE REVIEW

File: tests/snuba/api/endpoints/test_organization_events_spans_performance.py [hunk 1]
```diff
@@ -836,6 +887,48 @@ def test_group_filter(self, mock_raw_snql_query):
...
+            response = self.client.get(
+                self.url,
+                data={
+                    "project": self.project.id,
+                    "min_exclusive_time": 4,
+                },
+            )
+
+        assert response.status_code == 200, response.content
+        assert response.data == []
```
Comment (dashed) [inline tests/snuba/api/endpoints/test_organization_events_spans_performance.py:903]: @udameli These test cases can be more useful to assert that an event is within the filter bounds, and as well as asserting that an event is out of bounds wrt to the filters.

Comment (udameli) [reply tests/snuba/api/endpoints/test_organization_events_spans_performance.py:903]: updated!

**Review by dashed — APPROVED** [review]

**Comment by armenzg** [comment]
Notes to self.

This PR [skipped the steps](https://github.com/getsentry/sentry/runs/6097854000?check_suite_focus=true) for `webpack` [1].

On `master`, the steps were not skipped [2] and introduced [a regression](https://github.com/getsentry/sentry/actions/runs/2196922409).

It's failing with this [error](https://github.com/getsentry/sentry/runs/6098632872?check_suite_focus=true):
```
{
  "error": "SizeLimitError: Config option *webpack* needs *@size-limit/webpack* plugin\n    at checkChecks (/home/runner/work/sentry/sentry/node_modules/size-limit/get-config.js:60:17)\n    at getConfig (/home/runner/work/sentry/sentry/node_modules/size-limit/get-config.js:122:5)\n    at async module.exports (/home/runner/work/sentry/sentry/node_modules/size-limit/run.js:74:14)"
}
Error: Error parsing size-limit output. The output should be a json.
Error: TypeError: results.reduce is not a function
Error: results.reduce is not a function
```

Do you believe anything in this PR should have caused this error?

[1]
image

[2]
image

https://github.com/getsentry/sentry/actions/runs/2196922409

**Comment by armenzg** [comment]
The condition for [the steps](https://github.com/getsentry/sentry/blob/287b56ea4b0dbf22a6d70e160e74652358ec7d50/.github/workflows/frontend.yml#L158-L184) is `if: github.ref == 'refs/heads/master' || needs.files-changed.outputs.frontend == 'true'`.

The files-change job says that there are [no frontend changes](https://github.com/getsentry/sentry/runs/6097845921?check_suite_focus=true#step:3:68) in this PR.

The files in this PR touch the following files:

- src/sentry/api/endpoints/organization_events_spans_performance.py
- src/sentry/search/events/datasets/discover.py
- tests/snuba/api/endpoints/test_organization_events_spans_performance.py

and the rules to determine frontend changes makes sense: https://github.com/getsentry/sentry/blob/287b56ea4b0dbf22a6d70e160e74652358ec7d50/.github/file-filters.yml#L5-L27

**Comment by udameli** [comment]
@armenzg this PR on its own contained only backend changes so nothing frontend related should have been affected. One thing to note thought is that when I merged this PR to master it was a few commits behind master and those commits had some frontend changes

**Comment by armenzg** [comment]
> @armenzg this PR on its own contained only backend changes so nothing frontend related should have been affected. One thing to note thought is that when I merged this PR to master it was a few commits behind master and those commits had some frontend changes

Once merged, the size-limit-action will always run regardless of what changes were included. The condition says `if: github.ref == 'refs/heads/master' || needs.files-changed.outputs.frontend == 'true'`

**Pull request merged**